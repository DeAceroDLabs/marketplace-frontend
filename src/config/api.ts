import { API, Category, Product, ZipCodeLocation } from "./api.types";
import axios from "axios";
import { USER_API } from "./user.api.types";

const fetchData = (url: string) => {
  async function fetchUrl() {
    const response = await fetch(url);
    const json = await response.json();
    return json;
  }
  return fetchUrl();
};

const postRequest = async (url: string, body: any) => {
  const response = await axios.post(USER_API.createUser, body);
  return response;
};

const fetchCategories = (): Promise<Category[]> => {
  return fetchData(API.categories);
};

const fetchProducts = (categoryId: number): Promise<Product[]> => {
  return fetchData(API.products(categoryId));
};

const getLocation = (zipCode: string): Promise<ZipCodeLocation> => {
  return fetchData(API.location(zipCode)).then((data) => {
    if (data["location"]) {
      const neighborhoodsObject = data["location"]["neighborhood"];
      const neighborhoods = Object.keys(neighborhoodsObject);
      const neighborhoodsArray = neighborhoods.map((neighborhood) => {
        return {
          value: neighborhood,
          label: neighborhoodsObject[neighborhood],
        };
      });
      data["location"]["neighborhood"] = neighborhoodsArray;
    }
    return data;
  });
};

const searchProducts = (query: string): Promise<any> => {
  const body = {
    q: query,
  };
  return postRequest(API.search, body);
};

export {
  postRequest,
  fetchCategories,
  fetchProducts,
  searchProducts,
  getLocation,
};
