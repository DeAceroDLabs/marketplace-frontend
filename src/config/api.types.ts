export const API = {
  categories: "categories",
  products: (categoryId: number) => `categories/${categoryId}/products`,
  search: "/search",
  location: (zipCode: string) => `locations/postalCode/${zipCode}`,
};

export type Category = {
  categoryId: number;
  categoryName: string;
  imageUrl: string;
};

export type Article = {
  articleId: string;
};

export type Product = {
  productId: number;
  productName: string;
  categoryName: string;
  categoryId: number;
  articles: Article[];
  categoryImageUrl: string;
};

export type ZipCodeLocation = {
  countryName: string;
  location: Location;
};

export type Location = {
  cityId: string;
  city: string;
  stateId: string;
  state: string;
  neighborhood: Neighborhood[];
};

export type Neighborhood = {
  value: string;
  label: string;
};
