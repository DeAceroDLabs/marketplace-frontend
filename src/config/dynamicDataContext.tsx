import React from "react";

const defaultForm = {
  dynamicData: {} as any,
  setDynamicData: (data: any) => {},
};

export const DynamicDataContext = React.createContext(defaultForm);

export const DynamicDataProvider = DynamicDataContext.Provider;

export default DynamicDataContext;
