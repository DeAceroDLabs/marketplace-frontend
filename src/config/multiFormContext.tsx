import { Form } from "forms/form.types";
import React from "react";

const defaultForm = {
  forms: [] as Form[],
  activeForm: {} as Form,
  setActiveForm: (form: Form) => {},
};

const MultiFormContext = React.createContext(defaultForm);

export const MultiFormProvider = MultiFormContext.Provider;

export default MultiFormContext;
