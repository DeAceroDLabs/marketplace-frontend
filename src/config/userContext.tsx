import React from "react";

const defaultUser = {
  username: "",
  setUser: (username: string) => {},
};

const UserContext = React.createContext(defaultUser);

export const UserProvider = UserContext.Provider;

export default UserContext;
