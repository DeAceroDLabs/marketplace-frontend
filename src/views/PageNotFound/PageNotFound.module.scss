@use 'styles/colors';
@use 'styles/typography';

.info-container {
  height: 100px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}