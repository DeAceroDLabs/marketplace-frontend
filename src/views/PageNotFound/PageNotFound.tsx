import React from "react";
import NotFound from "components/common/NotFound";
import styles from "./PageNotFound.module.scss";
import Button from "components/common/Button";
import { useNavigate } from "react-router-dom";

const PageNotFound: React.FunctionComponent = () => {
  const navigate = useNavigate();
  return (
    <NotFound title="Página no encontrada" imgSrc="/page-not-found.jpg">
      <div className={styles["info-container"]}>
        <div>
          No hemos podido encontrar la página que buscas pero puedes seguir
          navegando en marketplace.
        </div>
        <Button color="primary-large" action={() => navigate("/")}>
          Ir a home
        </Button>
      </div>
    </NotFound>
  );
};

export default PageNotFound;
