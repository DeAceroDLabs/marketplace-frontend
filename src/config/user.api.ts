import { postRequest } from "./api";
import { CreateUser, USER_API } from "./user.api.types";

const createUser = (user: CreateUser) => {
  return postRequest(USER_API.createUser, user);
};
export { createUser };
