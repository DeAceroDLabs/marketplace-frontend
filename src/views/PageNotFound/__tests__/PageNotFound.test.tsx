import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import PageNotFound from "../PageNotFound";
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "config/userContext";

const mockUSer = {
  username: "user",
  setUser: jest.fn(),
};

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe("PageNotFound", () => {
  const renderView = () =>
    render(
      <UserProvider value={mockUSer}>
        <MemoryRouter>
          <PageNotFound />
        </MemoryRouter>
      </UserProvider>
    );
  it("renders PageNotFound without problem", async () => {
    expect(renderView()).toMatchSnapshot();
  });
  it("go to home button useNavigate", async () => {
    renderView();
    const getToKnowUsButton = screen.getByRole("button");
    fireEvent.click(getToKnowUsButton);
    await waitFor(() => {
      expect(mockedUsedNavigate).toBeCalled();
    });
  });
});
