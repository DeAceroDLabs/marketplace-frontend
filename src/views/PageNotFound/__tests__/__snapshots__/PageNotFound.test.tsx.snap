// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound renders PageNotFound without problem 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <img
            alt="Not found img"
            class="image"
            src="/page-not-found.jpg"
          />
          <div
            class="containerInfo"
          >
            <div
              class="title"
            >
              P치gina no encontrada
            </div>
            <div
              class="info"
            >
              <div
                class="info-container"
              >
                <div>
                  No hemos podido encontrar la p치gina que buscas pero puedes seguir navegando en marketplace.
                </div>
                <button
                  class="button primary-large activeClass"
                >
                  Ir a home
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <img
          alt="Not found img"
          class="image"
          src="/page-not-found.jpg"
        />
        <div
          class="containerInfo"
        >
          <div
            class="title"
          >
            P치gina no encontrada
          </div>
          <div
            class="info"
          >
            <div
              class="info-container"
            >
              <div>
                No hemos podido encontrar la p치gina que buscas pero puedes seguir navegando en marketplace.
              </div>
              <button
                class="button primary-large activeClass"
              >
                Ir a home
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
